python/ql/test/query-tests/Security/CWE-312-CleartextLogging/test.py

```diff
@@ -44,6 +44,37 @@ def print_password():
     x = getpass.getpass()
     print(x) # NOT OK
 
+def log_private():
+    def log1(social_security_number, ssn, className, passportNo):
+        print(social_security_number) # NOT OK
+        print(ssn) # NOT OK
+        print(className) # OK
+        print(passportNo) # NOT OK
+
+    def log2(post_code, zipCode, home_address):
+        print(post_code) # NOT OK, but NOT FOUND - "code" is treated as enxrypted and thus not sensitive
+        print(zipCode) # NOT OK, but NOT FOUND - "code" is treated as enxrypted and thus not sensitive
+        print(home_address) # NOT OK
+
+    def log3(user_latitude, user_longitude):
+        print(user_latitude) # NOT OK
+        print(user_longitude) # NOT OK
+
+    def log4(mobile_number, phoneNo):
+        print(mobile_number) # NOT OK
+        print(phoneNo) # NOT OK
+
+    def log5(creditcard, debit_card, bank_number, bank_account, accountNo, ccn, user_ccn, succNode):
+        print(creditcard) # NOT OK
+        print(debit_card) # NOT OK
+        print(bank_number) # NOT OK
+        print(bank_account) # NOT OK, but NOT FOUND - "account" is treated as having the "id" classification and thus excluded.
+        print(accountNo) # NOT OK, but NOT FOUND - "account" is treated as having the "id" classification and thus excluded.
```

Comment:
These `NOT FOUND` results are good observations.  We can think about mechanisms for limiting the exclusions perhaps and finding more results in cases like these.

Reply:
Should that be considered for this PR or as potential follow-up?

Reply:
As potential follow-up I'd say.  There will *always* be more improvements we can make to heuristics like this.